.JuliaFormatter.toml

```
@@ -0,0 +1,15 @@
+# Options for the JuliaFormatter auto syntax formatting tool.
+# https://domluna.github.io/JuliaFormatter.jl/stable/
+# https://docs.sciml.ai/SciMLStyle/stable/
+
+# We don't use SciML style since aligning with opening brackets doesn't look good if
+# the opening bracket is near the end of the line. This squeezes the code to the right,
+# and this alignment confuses auto indent.
+# Based on the default style we do pick these non-default options from SciML style:
+whitespace_ops_in_indices = true
+remove_extra_newlines = true
+always_for_in = true
+whitespace_typedefs = true
+
+# And add other options we like:
+separate_kwargs_with_semicolon = true
```

```suggestion
separate_kwargs_with_semicolon = true

ignore = [".pixi"]
```

This made it format the Ribasim repo on Windows in 1s instead of 7s.

I see you have julia installations in `utils/juliaup` as well, so you may need `ignore = [".pixi", "utils"]` (I did, until I got rid of that in https://github.com/Deltares/Ribasim/pull/1614. I think it'd be good for you to do the same, but can be a separate PR.)

---

How did you measure the formatting time?
I get less than 0.7 seconds doing this (without any of the pixi-related suggestions):

    julia> using BenchmarkTools

    julia> using JuliaFormatter

    julia> @btime format(".")
      672.831 ms (2384350 allocations: 143.38 MiB)

---

But is it faster with the ignore? I just did `@time format(".")`. There are actually two reasons for ignoring:

- We don't want to format any .jl files in there
- We don't want to spend time checking for tens of thousands of files if they are julia files or not

---

Be aware that by using `@time` the timing might include warm-up time (something you avoid with `BenchmarkTools.@btime`.
You might want to run `@time` twice to see the actual timing.

The ignore had no impact on my machine.

---

Yeah I'm aware. My timings were for Ribasim, where the pixi folder is much bigger, hence the effect more pronounced. I still see an improvement from 0.6s to 0.4s on my machine, which makes sense.